### PR TITLE
Log Missing Status Codes

### PR DESCRIPTION
**Pull request by anthony-murphy**
related to #4068

### CODE REVIEW

**Comment by anthony-murphy** [comment]
closed and re-opened to try and fix the auto assign job. seemed to work

File: packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts [hunk 1]
```diff
@@ -21,8 +21,8 @@ export enum R11sErrorType {
 function createNetworkError(
     errorMessage: string,
     canRetry: boolean,
-    statusCode?: number,
-    retryAfterSeconds?: number,
+    statusCode: number | undefined,
+    retryAfterSeconds: number | undefined,
```
Comment (markfields) [inline packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts:25]: Does anyone pass undefined now? Maybe just make it fully required as `number`?

Comment (markfields) [reply packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts:25]: And below in network.ts

**Review by markfields — APPROVED** [review]
🎂 LGTM

File: packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts]: can't change GenericeNetworkError due to createGenericNetworkError usages

**Comment by vladsud** [comment]
Is this change really needed? I prefer not to expose status codes, and rather make sure upper layers work with higher-level abstraction (errorType), which are unique to the point where we do not need fine details (of whether it was 401 or 403).
If we are missing details in the issue references here, then maybe getNackReconnectInfo() should do better job at projecting error codes to proper (unique) errorTypes (I'd say that ideally error codes should be removed from DeltaManager altogether and it's driver responsibility to do right things).

The reason I do not like having status codes is that app layers start to depend on them (we have to hide them to move them off status codes). And that makes code too dependent on particular storage / implementation details that may change in the future.

**Comment by anthony-murphy** [comment]
This is only logging. I think logging should always log the rawest data avaiable, otherwise it can hard to understand what happenning. I also agree we should do such detailed logic on status codes, or it should atleast be well encapsulated.

**Comment by anthony-murphy** [comment]
@vladsud just pushed a change that hopefully is more in line with what you want #4509

**Pull request merged**